src/windows/Button.tsx

```diff
@@ -0,0 +1,211 @@
+    private _isFocusedWithKeyboard = false;
+    private _isHoverStarted = false;
+
+    private _onFocusableRef = (btn: RNW.FocusableWindows | null): void => {
+        this._focusableElement = btn;
+    }
+
+    protected _render(internalProps: RN.ViewProps): JSX.Element {
+
+        // Fallback to native-common fast if the keyboard enabled component is not available
+        if (!IsUpdatedReactNativeForWindows) {
+            return super._render(internalProps);
+        }
+
+        // RNW.FocusableProps tabIndex: default is 0.
+        // -1 has no special semantic similar to DOM.
+        let tabIndex: number | undefined = this.getTabIndex();
+        // RNW.FocusableProps windowsTabFocusable:
+        // - true: keyboard focusable through any mean, receives keyboard input
+        // - false: not focusable at all, doesn't receive keyboard input
+        // The intermediate "focusable, but not in the tab order" case is not supported.
+        let windowsTabFocusable: boolean = !this.props.disabled && tabIndex !== undefined && tabIndex >= 0;
+
+        // RNW.FocusableWindows doesn't participate in layouting, it basically mimics the position/size of the child
+        let focusableViewProps: RNW.FocusableProps = {
+            ref: this._onFocusableRef,
+            isTabStop: windowsTabFocusable,
+            tabIndex: tabIndex,
+            disableSystemFocusVisuals: false,
+            handledKeyDownKeys: DOWN_KEYCODES,
+            handledKeyUpKeys: UP_KEYCODES,
+            onKeyDown: this._onKeyDown,
+            onKeyUp: this._onKeyUp,
+            onFocus: this._onFocus,
+            onBlur: this._onBlur
+        };
+
+        return (
+            <RNW.FocusableWindows
+                {...focusableViewProps}
+            >
+                <RN.Animated.View
+                    {...internalProps}
+                    onMouseEnter={this._onMouseEnter}
+                    onMouseLeave={this._onMouseLeave}
+                >
+                    { this.props.children }
+                </RN.Animated.View>
+            </RNW.FocusableWindows>
+        );
+    }
+
+    focus() {
+        super.focus();
+        if (this._focusableElement && this._focusableElement.focus) {
+            this._focusableElement.focus();
+        }
+
+    }
+
+    blur() {
+        super.blur();
+        if (this._focusableElement && this._focusableElement.blur) {
+            this._focusableElement.blur();
+        }
+    }
+
+    private _onKeyDown = (e: React.SyntheticEvent<any>): void => {
+
+        if (!this.props.disabled) {
+            let keyEvent = EventHelpers.toKeyboardEvent(e);
+            if (this.props.onKeyPress) {
+                this.props.onKeyPress(keyEvent);
+            }
+
+            if (this.props.onPress) {
+                let key = keyEvent.keyCode;
+                // ENTER triggers press on key down
+                if (key === KEY_CODE_ENTER) {
+                    this.props.onPress(keyEvent);
```

Comment:
>keyEvent [](start = 39, length = 8)

Is this the type of event this handler expects?

Reply:
This handler expects less (SyntheticEvent), but sine we already have the beefed up event, we pass that one.

---
In reply to: [159338363](https://github.com/Microsoft/reactxp/pull/404#discussion_r159338363) [](ancestors = 159338363)